### ISSUE TITLE
Regent: typechecking issue in print statement

### ISSUE THREAD

**Issue by LonelyCat124** [StanfordLegion/legion#837]
When I try to run the following program, I get a typechecking issue. With the **d lines removed the program works correctly:
error:
```
../../legion/language/regent.py test2.rg
...e/aidan/Legion/legion/language/src/regent/symbol_table.t:66: Errors reported during typechecking.
test2.rg:38: name '$85' is already defined
  c.printf("ispace = %i\n", f[0].ispace.volume)
                             ^

stack traceback:
        [C]: in function 'error'
        /home/aidan/Legion/legion/language/terra/src/terralib.lua:390: in function 'finishandabortiferrors'
        /home/aidan/Legion/legion/language/src/common/report.t:56: in function 'error'
        ...e/aidan/Legion/legion/language/src/regent/symbol_table.t:66: in function 'insert'
        /home/aidan/Legion/legion/language/src/regent/validate.t:57: in function 'intern_variable'
        /home/aidan/Legion/legion/language/src/regent/validate.t:270: in function 'fn'
        /home/aidan/Legion/legion/language/src/common/ast.t:426: in function 'continuation'
        /home/aidan/Legion/legion/language/src/common/ast.t:436: in function 'continuation'
        /home/aidan/Legion/legion/language/src/common/ast.t:431: in function 'continuation'
        /home/aidan/Legion/legion/language/src/regent/validate.t:92: in function 'fn'
        /home/aidan/Legion/legion/language/src/common/ast.t:426: in function 'continuation'
        /home/aidan/Legion/legion/language/src/common/ast.t:440: in function 'traverse_node_continuation'
        /home/aidan/Legion/legion/language/src/regent/validate.t:302: in function 'validate_variables'
        /home/aidan/Legion/legion/language/src/regent/validate.t:310: in function 'top_task'
        /home/aidan/Legion/legion/language/src/regent/validate.t:315: in function </home/aidan/Legion/legion/language/src/regent/validate.t:313>
        /home/aidan/Legion/legion/language/src/regent/passes.t:44: in function </home/aidan/Legion/legion/language/src/regent/passes.t:40>
        test2.rg:10: in main chunk
```

Minimal program:

```
import "regent"

local c = regentlib.c

fspace cell{
  x : double,
  neigh_bound : int1d
}

task main()
  var num_elements = 5
  var is = ispace(int2d, {x = num_elements+2, y = num_elements+2}, {x = -1, y = -1})
  var solution = region(is, cell)
  fill(solution.x, 0.0)

  var xlo = solution.bounds.lo.x
  var xhi = solution.bounds.hi.x
  var ylo = solution.bounds.lo.y
  var yhi = solution.bounds.hi.y
  for point in solution do
    if point.x >= 0 and point.y >= 0 and point.x < xhi and point.y < yhi then
      solution[point].x = point.x*point.y
    else
      solution[point].x = -1
    end
  end

  for point in solution do
    if( solution[point+{-1,0}].x < 0 or solution[point + {1,0}].x < 0 or solution[point+{0,1}].x < 0 or solution[point + {0,-1}].x < 0 or solution[point].x < 0 ) then
      solution[point].neigh_bound = 1
    end
  end

  var f = partition(solution.neigh_bound, ispace(int1d, 2))
  ** for s in f[0].ispace do **
    ** c.printf("%i %i\n", s.x, s.y) **
  ** end **
  c.printf("ispace = %i\n", f[0].ispace.volume)
end

regentlib.start(main)
```

Current master, compiled with `./install.py --clean`

**Comment by elliottslaughter** [comment]
Looks like a compiler issue, I'll try to investigate.

**Comment by rupanshusoi** [comment]
Attaching a smaller reproducer
```
import "regent"
task foo(x : ispace(int1d)) end
task main()
  var r = region(ispace(int1d, 10), int)
  var p = partition(equal, r, ispace(int1d, 2))
  foo(p[0].ispace)
  foo(p[0].ispace)
end
regentlib.start(main)
```

**Comment by elliottslaughter** [comment]
Just FYI, @rupanshusoi noticed that the problem goes away if you use `-fflow 0`, so you can use that as a workaround until we get this fixed.

**Comment by elliottslaughter** [comment]
Will be resolved by #1528.

**Comment by elliottslaughter** [comment]
SCR has been removed.
